src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -911,10 +915,70 @@ protected override void AddExistingDocument(DocumentId documentId, string filePa
                 throw new Exception(ServicesVSResources.Could_not_find_location_of_folder_on_disk);
             }
 
-            return AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+            AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+        }
+
+        private bool TryAddEditorConfigToSolutionItems(
+            DocumentInfo documentInfo,
+            SourceText text)
+        {
+            // We're going to see if this is an .editorconfig being added at the solution level, and if so process it specially; so verify first that it is one
+            if (CurrentSolution.FilePath is null || documentInfo.FilePath is null)
+                return false;
+
+            if (PathUtilities.GetFileName(documentInfo.FilePath) != ".editorconfig")
+                return false;
+
+            if (IOUtilities.PerformIO(() => File.Exists(documentInfo.FilePath)))
+                return false;
+
+            var solutionDirectory = PathUtilities.GetDirectoryName(this.CurrentSolution.FilePath);
+            if (solutionDirectory != PathUtilities.GetDirectoryName(documentInfo.FilePath))
+                return false;
+
+            // Double check too that this isn't a case of the .csproj being in the same folder of the .sln, at which point it's reasonable
+            // just to add this to the project file.
+            if (PathUtilities.GetDirectoryName(CurrentSolution.GetProject(documentInfo.Id.ProjectId)?.FilePath) == solutionDirectory)
+                return false;
+
+            // All checks pass, so let's treat this special.
+            var dte = _threadingContext.JoinableTaskFactory.Run(() => _asyncServiceProvider.GetServiceAsync<SDTE, EnvDTE.DTE>(_threadingContext.JoinableTaskFactory));
+
+            const string SolutionItemsFolderName = "Solution Items";
```

Comment:
is this ok in a localized world?

Reply:
if so, doc.

Reply:
So this is preserving the existing behavior of this code if you compare to what was deleted. Digging a bit further yes, we perhaps should be considering localization here, since the solution items folder absolutely can be renamed, and adding a solution-level text file indeed picks a different name if you're on a different language:

<img width="263" alt="image" src="https://github.com/dotnet/roslyn/assets/201340/8925b5c3-f408-4881-bc4d-4828208948de">

I'll file a follow-up item to investigate what the right change is here. We could easily remove the check for folder name (and just use any existing solution items folder that's there), but that still raises the question for what we have to pass when creating one. And that's not something we'll want to touch for 17.9 anyways, since this is not a regression.

Reply:
Filed https://github.com/dotnet/roslyn/issues/71673